freud/common.py

```diff
@@ -56,3 +56,246 @@ def convert_box(box):
+        canvas = FigureCanvasAgg(fig) # noqa F841
+        fig.savefig(f, format='png')
+        return f.getvalue()
+
+
+def line_plot(x, y, title=None, xlabel=None, ylabel=None):
+    """ Helper function to draw a line graph.
+
+    .. moduleauthor:: Jin Soo Ihm <jinihm@umich.edu>
+
+    Args:
+        x (list): x values of the line graph.
+        y (list): y values corresponding to :code:`x`.
+        title (str): Title of the graph. (Default value = :code:`None`).
+        xlabel (str): Label of x axis. (Default value = :code:`None`).
+        ylabel (str): Label of y axis. (Default value = :code:`None`).
+
+    Returns:
+        bytes: Byte representation of the graph in png file if import
+            succeeds. Otherwise :code:`None`.
+    """
+    try:
+        from matplotlib.figure import Figure
+        from matplotlib.backends.backend_agg import FigureCanvasAgg
+        import io
+    except ImportError:
+        return None
+    else:
+        fig = Figure()
+        ax = fig.subplots()
+        ax.plot(x, y)
+        ax.set_title(title)
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+        f = io.BytesIO()
+        canvas = FigureCanvasAgg(fig) # noqa F841
+        fig.savefig(f, format='png')
+        return f.getvalue()
+
+
+def make_polygon(sides, radius=1):
+    """ Helper function to draw polygon taken from freud-exapmles.
+    """
+    thetas = np.linspace(0, 2*np.pi, sides+1)[:sides]
+    vertices = np.array([[radius*np.sin(theta), radius*np.cos(theta)]
+                         for theta in thetas])
+    return vertices
+
+
+def pmft_plot(pmft):
+    """ Helper function to draw 2D PMFT diagram.
+
+    Args:
+        pmft (:class:`freud.pmft.PMFTXY2D`):
+            PMFT2DXY instance.
+
+    Returns:
+        bytes: Byte representation of the diagram in png file if import
+            succeeds. Otherwise :code:`None`.
+    """
+    try:
+        from matplotlib.figure import Figure
+        from matplotlib.backends.backend_agg import FigureCanvasAgg
+        from matplotlib.colors import Normalize
+        from matplotlib.cm import viridis
+        import io
+        from scipy.ndimage.filters import gaussian_filter
+    except ImportError:
+        return None
+    else:
+        pmft_arr = np.copy(pmft.PMFT)
+
+        # Do some simple post-processing for plotting purposes
+        pmft_arr[np.isinf(pmft_arr)] = np.nan
+        dx = (2.0 * 3.0) / pmft.n_bins_X
+        dy = (2.0 * 3.0) / pmft.n_bins_Y
+        nan_arr = np.where(np.isnan(pmft_arr))
+        for i in range(pmft.n_bins_X):
+            x = -3.0 + dx * i
+            for j in range(pmft.n_bins_Y):
+                y = -3.0 + dy * j
+                if ((x*x + y*y < 1.5) and (np.isnan(pmft_arr[j, i]))):
+                    pmft_arr[j, i] = 10.0
+        w = int(2.0 * pmft.n_bins_X / (2.0 * 3.0))
+        center = int(pmft.n_bins_X / 2)
+
+        # Get the center of the histogram bins
+        pmft_smooth = gaussian_filter(pmft_arr, 1)
```

Comment:
I'm not sure if smoothing is appropriate for displaying the raw data. Taking this out would also remove a dependency.